docs/examples/runtimeDashboard.md

```diff
@@ -153,11 +95,57 @@ The 'GC Duration Mean' panel extends the `base` panel to display the duration in
 }
 ```
 
-### Queries
+### Rows and Grid
+
+Rows can be used to group panels and optionally collapse them, however using them within Jsonnet can be quite cumbersome. For example: all panels added after a row in the panel array will inevitably become part of the row, also the order in the array doesn't necessarily apply with how Grafana displays them. The `makeGrid` util function attempts to aid with this.
+
+First the panels are consistently added to the rows so that the intention is clear from a Jsonnet perspective.
+
+```jsonnet
+    row.new('Process')
+    + row.withPanels([
+      panels.timeSeries.cpuUsage('CPU Usage', queries.cpuUsage),
+      panels.timeSeries.memoryUsage('Memory Usage', queries.memUsage),
+      panels.timeSeries.base('Goroutines', queries.goroutines),
+      panels.timeSeries.base('Threads', queries.threads),
+      panels.timeSeries.seconds('GC Duration Mean', queries.gcDuration),
+    ]),
+```
+
+Second the panel array gets processed by `makeGrid` before added it to the dashboard.
+
+```jsonnet
++ g.dashboard.withPanels(
+  g.util.grid.makeGrid([
+    row.new('Process')
+    + row.withPanels([
+      panels.timeSeries.cpuUsage('CPU Usage', queries.cpuUsage),
+      panels.timeSeries.memoryUsage('Memory Usage', queries.memUsage),
+      panels.timeSeries.base('Goroutines', queries.goroutines),
+      panels.timeSeries.base('Threads', queries.threads),
+      panels.timeSeries.seconds('GC Duration Mean', queries.gcDuration),
+    ]),
+    row.new('Kubernetes Client')
+    + row.withCollapsed(true)
+    + row.withPanels([
+      panels.heatmap.base('Workqueue Waiting Duration Over Time', queries.wqDurationOverTime),
+      panels.timeSeries.durationQuantile('Workqueue Waiting Duration Quantile', queries.wqDurationQuantile),
+      panels.timeSeries.short('Workqueue Depth', queries.wqDepth),
+      panels.timeSeries.short('Failed Requests', queries.failedRequests),
+    ]),
+    row.new('Controller Runtime')
+    + row.withCollapsed(true)
+    + row.withPanels([
+      panels.heatmap.base('Reconciling Latency Over Time', queries.reconcilingLatencyOverTime),
+      panels.timeSeries.durationQuantile('Reconciling Latency Quantile', queries.reconcilingDurationQuantile),
+    ]),
```

Comment:
Looks like there's a bit missing here (closing the `makeGrid()` and `withPanels()` calls)

Reply:
good catch, fixed